packages/component-library/src/components/form/mt-unit-field/mt-unit-field.spec.ts

```diff
@@ -0,0 +1,107 @@
+import { render, screen } from "@testing-library/vue";
+import MtUnitField from "./mt-unit-field.vue";
+import { userEvent } from "@testing-library/user-event";
+import type { Unit } from "convert-units";
+
+describe("mt-unit-field", () => {
+  it("is not possible to change the value when inheritance is linked", async () => {
+    // ARRANGE
+    const handler = vi.fn();
+
+    render(MtUnitField, {
+      props: {
+        modelValue: 0,
+        defaultUnit: "mm" as Unit,
+        isInheritanceField: true,
+        isInherited: true,
+        "onUpdate:modelValue": handler,
+      },
+    });
+
+    // ACT
+    await userEvent.type(screen.getByRole("textbox"), "1");
+
+    // ASSERT
+    expect(screen.getByRole("textbox")).toBeDisabled();
+    expect(screen.getByRole("textbox")).toHaveValue("0");
+    expect(handler).not.toHaveBeenCalled();
+  });
+
+  it("converts value when unit is changed", async () => {
+    // ARRANGE
+    const modelValueHandler = vi.fn();
+    const unitHandler = vi.fn();
+
+    render(MtUnitField, {
+      props: {
+        modelValue: 1,
+        defaultUnit: "m" as Unit,
+        measurementType: "length",
+        digits: 3,
+        "onUpdate:modelValue": modelValueHandler,
+        "onUpdate:defaultUnit": unitHandler,
+      },
+    });
+
+    // ACT
+    const unitSelect = screen.getByTestId("unit-select-trigger");
+    await userEvent.click(unitSelect);
+
+    const ydOption = screen.getByTestId("unit-select-option-yd");
+    expect(ydOption).toBeInTheDocument();
+    await userEvent.click(ydOption);
+
+    // ASSERT
+    expect(modelValueHandler).toHaveBeenCalledWith(1.094); // 1 m = 1.094 yd
+    expect(unitHandler).toHaveBeenCalledWith("yd");
+  });
+
+  it("displays different units when property measurementType is changed", async () => {
+    // ARRANGE
+    render(MtUnitField, {
+      props: {
+        modelValue: 1,
+        defaultUnit: "g" as Unit,
```

Comment:
Is the type cast necessary? If yes we need to change that. Otherwise, developers have to install the convert-unit package themselves too.